src/HelperLocator.php

```diff
@@ -3,68 +3,27 @@
-            'j'                     => function () use ($escape) { return new Helper\EscapeJs($escape); },
-            'label'                 => function () use ($escape) { return new Helper\Label($escape); },
-            'link'                  => function () use ($escape) { return new Helper\Link($escape); },
-            'linkStylesheet'        => function () use ($escape) { return new Helper\LinkStylesheet($escape); },
-            'meta'                  => function () use ($escape) { return new Helper\Meta($escape); },
-            'metaHttp'              => function () use ($escape) { return new Helper\MetaHttp($escape); },
-            'metaName'              => function () use ($escape) { return new Helper\MetaName($escape); },
-            'monthField'            => function () use ($escape) { return new Helper\MonthField($escape); },
-            'numberField'           => function () use ($escape) { return new Helper\NumberField($escape); },
-            'ol'                    => function () use ($escape) { return new Helper\Ol($escape); },
-            'passwordField'         => function () use ($escape) { return new Helper\PasswordField($escape); },
-            'radioField'            => function () use ($escape) { return new Helper\RadioField($escape); },
-            'rangeField'            => function () use ($escape) { return new Helper\RangeField($escape); },
-            'resetButton'           => function () use ($escape) { return new Helper\ResetButton($escape); },
-            'script'                => function () use ($escape) { return new Helper\Script($escape); },
-            'searchField'           => function () use ($escape) { return new Helper\SearchField($escape); },
-            'select'                => function () use ($escape) { return new Helper\Select($escape); },
-            'submitButton'          => function () use ($escape) { return new Helper\SubmitButton($escape); },
-            'telField'              => function () use ($escape) { return new Helper\TelField($escape); },
-            'textarea'              => function () use ($escape) { return new Helper\Textarea($escape); },
-            'textField'             => function () use ($escape) { return new Helper\TextField($escape); },
-            'timeField'             => function () use ($escape) { return new Helper\TimeField($escape); },
-            'u'                     => function () use ($escape) { return new Helper\EscapeUrl($escape); },
-            'ul'                    => function () use ($escape) { return new Helper\Ul($escape); },
-            'urlField'              => function () use ($escape) { return new Helper\UrlField($escape); },
-            'weekField'             => function () use ($escape) { return new Helper\WeekField($escape); },
-        ]);
+        return new static([
+            'a'      => static fn () => new Helper\EscapeAttr($escape),
```

Comment:
You are explicitly defining this because they are not having the same name. I guess this will be also the situation of plugins. May be I would prefer passing extra factories to the new method there by merging the default and the newly passed ones there by overriding the default ones itself.

Also there is an advantage because the `$name` currently will be smallcase. Current php version doesnot have any problem with creating objects if class is `Example` and calling `new example()` . But I don't know about the future what will happen to those. 

I don't like this approach. ie just my opinion.

But good to see a different implementation and different thinking.

Reply:
@harikt It's easy to align class names and helper names when they have conventions (just don't implement them that way because you don't need to) I'd be happy to focus on what problems to solve, not how the implementation is done. What about sharing and reuse?

Reply:
@koriym sharing and reuse that is a good topic. Current implementation supports it. But there is a little problem I see how new helper methods are added. So I have proposed an alternative way https://github.com/qiqphp/qiq/pull/9 . The only thing is the developer need to add the factories explicitly.

I just shared my views, different people have different conventions.

Reply:
@harikt Thank you for sharing your thought and implementation. I prefer lazy-load plugin style rather than a service locator that packs everything into a container. However, the original implementation was clearly the latter approach, and your new PR follows it.

Reply:
@harikt Note that this PR adds a plugin style for lazy loading, but does not negate or invalidate the existing approach.